test/ziggurat/streams_test.clj

```diff
@@ -102,6 +103,67 @@
     (stop-streams streams)
     (is (= times @message-received-count))))
 
+(deftest stop-stream-test
+  (let [message-received-count (atom 0)
+        mapped-fn              (get-mapped-fn message-received-count)
+        times                  6
+        kvs                    (repeat times message-key-value)
+        handler-fn             (default-middleware/protobuf->hash mapped-fn proto-class :default)
+        _                      (mount/start)
+        streams                (start-streams {:default {:handler-fn handler-fn}}
+                                              (-> (ziggurat-config)
+                                                  (assoc-in [:stream-router :default :application-id] (rand-application-id))
+                                                  (assoc-in [:stream-router :default :changelog-topic-replication-factor] changelog-topic-replication-factor)))]
+    (Thread/sleep 10000)                                    ;;waiting for streams to start
+    (IntegrationTestUtils/produceKeyValuesSynchronously (get-in (ziggurat-config) [:stream-router :default :origin-topic])
+                                                        kvs
+                                                        (props)
+                                                        (MockTime.))
+    (Thread/sleep 10000)                                     ;;wating for streams to consume messages
+    (stop-stream :default)
+    (is (= times @message-received-count))))
```

Comment:
How is this assertion help us know that the stream was indeed closed properly?

Shouldn't you redef `.close` method on streams and update an atom to true

Please look at other tests in Ziggurat test dir for examples on how to do this.

Reply:
Instead, as @macalimlim suggested, we could check the state of the streams after stopping the streams using the `.start` method provided to us.